lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -224,9 +249,26 @@ public APIGatewayProxyResponseEvent handleRequest(
             LogHelper.logErrorMessage("Failed to parse request body into map.", e.getMessage());
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_IP_ADDRESS);
+        } catch (VerifiableCredentialException e) {
+            LogHelper.logErrorMessage(
+                    "Inherited identity validation failed.", e.getErrorResponse().getMessage());
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatus.SC_BAD_REQUEST, ErrorResponse.INVALID_SESSION_REQUEST);
         }
     }
 
+    private void validateAndStoreHMRCInheritedIdentity(String inheritedIdentityJWT, String userId)
+            throws ParseException, VerifiableCredentialException {
+        SignedJWT signedInheritedIdentityJWT = SignedJWT.parse(inheritedIdentityJWT);
+        CriConfig inheritedIdentityCriConfig = configService.getCriConfig(HMRC_MIGRATION_CRI);
+
+        verifiableCredentialJwtValidator.validate(
+                signedInheritedIdentityJWT, inheritedIdentityCriConfig, userId);
+
+        verifiableCredentialService.persistUserCredentials(
+                signedInheritedIdentityJWT, HMRC_MIGRATION_CRI, userId);
```

Comment:
I'm going to hazard a guess we're going to need a new audit event in here.

Reply:
Audit event is covered by https://govukverify.atlassian.net/browse/PYIC-4248 (or perhaps one of the associated tickets)